### ISSUE TITLE
Font metadata read incorrectly in OSD firmware v2.0.0

### ISSUE THREAD

**Issue by pawelsky** [FrSkyRC/PixelOSD#17]
After uploading the font using the FrSky OSD App v2.0.2 to the OSD running firmware v 2.0.0 the font metadata is stored incorrectly. 

I'm using the [vision.mcm](https://github.com/iNavFlight/inav-configurator/blob/master/resources/osd/vision.mcm) font with the [vision.yaml](https://github.com/iNavFlight/inav-configurator/blob/master/resources/osd/vision.yaml) metadata.

When I read the character #150 from the OSD I'm getting the following (incorrect) metadata
`82 5 4 1 4 101 0 85 97 2`

When I load the exact same font in the simulator v1.0.3 and read the metadata of character #150 it is correct
`114 5 4 3 12 111 0 255 99 2`

**Comment by fiam** [comment]
@pawelsky Sorry for not replying sooner. I've been swamped with other projects lately. I've been trying to reproduce this for several weeks with no luck. Could you please try it with this beta build for the firmware? 

[FrSkyOSD-v2.0.1_20201017.bin.zip](https://github.com/FrSkyRC/PixelOSD/files/5395941/FrSkyOSD-v2.0.1_20201017.bin.zip)

You can flash it with either the Python SDK (`python3 frskyosd.py --flash file.bin`) or with the OSD app (using the `Select File` button after clicking on `Flash Firmware`).

I've been using this Python script for testing. I'm attaching it in case it can be helpful. Note that with all the released versions there's a bug in `OSD_CMD_WRITE_FONT` that returns invalid data in its response (but the correct data should be still stored in the OSD anyway). It's fixed in this beta version.


[font.py.zip](https://github.com/FrSkyRC/PixelOSD/files/5395964/font.py.zip)

**Comment by fiam** [comment]
Forgot to mention what I've tried so far. This is what I get with the `font.py` script after uploading `vision.mcm`:

```
[fiam@muk:sdk/python]$ python3 font.py /dev/tty.SLAB_USBtoUART --read-char 150                                                                                                                (10-17 16:45)
FrSky OSD 2.0.1, 30x16 grid, 360x288 pixels
150 CMD = 2, 66 bytes = 960055555555555555001554aa8554808554808554808554808554808554808554808554808554808554808554808554808554aa85550015720504030c6f00ff6302
```

Note the last 10 bytes in decimal are `114 5 4 3 12 111 0 255 99 2`, which correspond to:

```
72 = 114 = 'r'
05 = 5
04 = 4
03 = 3
0c = 12
6f = 111 = 'f'
00 = 0
ff = int8(-1)
63 = 99 = 'c'
02 = 0b10 = WHITE, BLACK
```


I've tried uploading random font data, uploading fonts directly, uploading fonts from INAV configurator. It always stores the correct data.

**Comment by pawelsky** [comment]
Hi, thanks for feedback - I did a bit more digging and it looks like this is not a problem of the OSD firmware but the FrSky OSD App that uploads font to the OSD. I've captured the data that is being sent via the COM port and it is exactly what I'm reading (i.e. the incorrect values)

![image](https://user-images.githubusercontent.com/7121220/96351474-ce3d8880-10bb-11eb-8334-21f6274fe3b2.png)

Here is output from your font.py after uploading the font. As expected it is also incorrect

```
FrSky OSD 2.0.1, 30x16 grid, 360x288 pixels
150 CMD = 2, 66 bytes = 960055555555555555001554aa8554808554808554808554808554808554808554808554808554808554808554808554808554aa8555001552050401046500556102
```

The data in the font file I'm uploading is correct

![image](https://user-images.githubusercontent.com/7121220/96351532-33917980-10bc-11eb-816d-db0ca65a3144.png)

What do you use to upload font to OSD?

**Comment by pawelsky** [comment]
Did another test this time uploading the same font using frskyosd.py.

The COM port communication seems correct

![image](https://user-images.githubusercontent.com/7121220/96354728-123f8600-10da-11eb-93ee-c042684cebce.png)

And the output from the font.py is also correct this time

```
FrSky OSD 2.0.1, 30x16 grid, 360x288 pixels
150 CMD = 2, 66 bytes = 960055555555555555001554aa8554808554808554808554808554808554808554808554808554808554808554808554808554aa85550015720504030c6f00ff6302
```
This proves that the problem is in the FrSkyOSD App - can you move this issue to the other repo?

**Comment by fiam** [comment]
Thank you so much for your help @pawelsky. I've reproduced this locally and found and fixed the bug. I'm preparing a new app release as we speak.

Turns out the bug was here https://github.com/fiam/max7456tool/blob/cbc9b08c76bff70daf1e2be15882d95b5fd7d2f6/mcm/char_builder.go#L31

I wrote that code several years ago, before designing an OSD had ever crossed my mind, and I did so following MAX7456 data sheet. Turns out, both BF and INAV had at that time `.mcm` files maintained by hand, and there were some files with pixels set to `11`, which is an invalid value (valid ones for MAX7456 are `00` for black, `10` for white and `01` for transparent). This worked fine on the original MAX756 (it interpreted them as transparent), but in some of the early clone chips they interpreted it as something else (can't remember if black or white). The obvious solution at the time was to make the decoder fix up the data as it was reading it and regenerate the files. Now that we have OSDs that both understand `11` as gray and use metadata, that's definitely a problem. Since we're no longer using `.mcm` as the canonical format for fonts in INAV either (I migrated it to `.png` some years ago), we don't need to mangle the `.mcm` data anymore, so at this point the mangling can be safely removed.

Again, thank you so much for your invaluable help and patience.
